transhumandesign/kag-base
# [fixed] "on fire" state wasn't copied when changing class

## Status

- **READY**: this PR is (to the best of your knowledge) ready to be incorporated into the game.

## Description

```
[fixed/added] "on fire" will be copied when changing class
```
Fixes https://github.com/transhumandesign/kag-base/issues/1300

This PR makes it so that changing class will not get rid of you being on fire. Instead your fire state will be copied to the new blob.

Because `this.getCurrentScript().runFlags` etc. would cause an exception when the script it's in is being used as an "include script" rather than a "blob script", I removed it and made `IsFlammable.as`'s `onTick()` only run when the condition `!this.hasTag(burning_tag) && !this.isInFlames()` is false.

Tested online and offline and it works as it should.

## Steps to Test or Reproduce

Spawn a tdm_spawn.
Be archer.
Shoot fire arrow into the grass and wait for it to burn and set you on fire.
Run into the tdm_spawn and change class.
Notice the burning will keep going as long as it would have without changing class.
